lib/ansible/playbook/role/definition.py

```diff
@@ -39,11 +41,11 @@
 display = Display()
 
 
-class RoleDefinition(Base, Become, Conditional, Taggable):
+class RoleDefinition(Base, Become, Conditional, Taggable, Collection):
 
     _role = FieldAttribute(isa='string')
 
-    def __init__(self, play=None, role_basedir=None, variable_manager=None, loader=None):
+    def __init__(self, play=None, role_basedir=None, variable_manager=None, loader=None, collection_list=[]):
```

Comment:
Can we please use some immutable default value?
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

```suggestion
    def __init__(self, play=None, role_basedir=None, variable_manager=None, loader=None, collection_list=None):
```

It's fine to assign `[]` inside the constructor, just not in the definition.